webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModalActions.js

```diff
@@ -2,49 +2,48 @@ import {
   ADD_MODAL,
   SET_MODAL_OPEN,
   SET_MODAL_CLOSED,
+  SET_MODAL_START_SUBMITTING,
+  SET_MODAL_STOP_SUBMITTING,
 } from './ForemanModalConstants';
 import { selectModalStateById } from './ForemanModalSelectors';
 
-export const addModal = ({ id, open = false }) => (dispatch, getState) => {
+export const addModal = ({ id, open = false, submitting = false }) => (
+  dispatch,
+  getState
+) => {
   const modalAlreadyExists = selectModalStateById(getState(), id);
   if (modalAlreadyExists) {
     throw new Error(`ForemanModal with ID ${id} already exists`);
   }
   return dispatch({
     type: ADD_MODAL,
-    payload: { id, open },
+    payload: { id, open, submitting },
   });
 };
 
-export const setModalOpen = ({ id }) => (dispatch, getState) => {
+const modalAction = actionType => ({ id }) => (dispatch, getState) => {
```

Comment:
Nice elegant way to expand the checking for existence of a modal to multiple actions.  I was actually thinking that throwing errors for this is too annoying for developers anyway, and that the fact that we use constants for modal ID's is enough to prevent modal ID confusion.  I may put in a PR soon to remove the error throwing (just FYI).  thoughts welcome